lib/spack/spack/install_test.py

```diff
@@ -3,32 +3,73 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 import base64
+import contextlib
+import enum
 import hashlib
+import inspect
+import io
 import os
 import re
 import shutil
+import sys
+from collections import Counter
+from typing import Callable, List, Optional, Tuple, Type, Union
 
 import llnl.util.filesystem as fs
 import llnl.util.tty as tty
+from llnl.util.lang import nullcontext
+from llnl.util.tty.color import colorize
 
 import spack.error
 import spack.paths
 import spack.util.spack_json as sjson
 from spack.spec import Spec
 from spack.util.prefix import Prefix
+from spack.util.string import plural
 
+#: Stand-alone test failure info type
+TEST_FAILURE_TYPE = Tuple[BaseException, str]
+
+#: Name of the test suite's (JSON) lock file
 test_suite_filename = "test_suite.lock"
+
+#: Name of the test suite results (summary) file
 results_filename = "results.txt"
 
+#: Name of the Spack install phase-time test log file
+spack_install_test_log = "install-time-test-log.txt"
+
+
+ListOrStringType = Union[str, List[str]]
+LogType = Union["tty.log.nixlog", "tty.log.winlog"]
+SpecType = List[Spec]
+
+
+@enum.unique
+class TestStatus(enum.Enum):
+    """Names of different stand-alone test states."""
```

Comment:
Do we use `unique` because we rely on the actual value somewhere?

Reply:
Good point.  I don't think so but will check.

Reply:
Looks like I made a point of using the name.